drivers/usb/gadget/xaptum/f_psock_gadget.c

```diff
@@ -302,7 +302,14 @@ static void psock_free_func( struct usb_function *f )
         mutex_unlock(&opts->lock);
 
         usb_free_all_descriptors(f);
-        kfree(func_to_psock(f));
+        if(!func_to_psock(f))
```

Comment:
`func_to_psock` should never return null --- it's just a simple lookup via `container_of`.  If it does, there's a bug where the function and psock structs are initialized. Let's fix it there, rather than masking the bug here.

Reply:
I don't think there was ever a memory leak here, this is probably just a test I created and lost track of.